admin/src/main/scala/io/buoyant/admin/PrometheusStatsHandler.scala

```diff
@@ -22,21 +22,17 @@ private[admin] object PrometheusStatsHandler {
   private[this] case class Escape(regex: Regex, replace: String) {
     private[admin] def replaceAllIn(str: String) = regex.replaceAllIn(str, replace)
   }
-  private[this] val escape1 = Escape("[!,]".r, "-")
-  private[this] val escape2 = Escape("[/]".r, ":")
-  private[this] val statPattern = """(.*)\.(count|sum|avg|min|max|stddev|p50|p90|p95|p99|p9990)$""".r
-
-  private[this] val escape3 = Escape("[.\\-$]".r, "_")
+  private[this] val delimiter = Escape("[/]".r, ":")
+  private[this] val statPattern = """(.*)_(count|sum|avg|min|max|stddev|p50|p90|p95|p99|p9990)$""".r
+  private[this] val disallowedChars = Escape("[^a-zA-Z0-9_:]".r, "_")
 
   private[admin] def formatKey(key: String) = {
-    val escapedKey = escape2.replaceAllIn(escape1.replaceAllIn(key))
-
+    val escapedKey = disallowedChars.replaceAllIn(delimiter.replaceAllIn(key))
     val parsedKey = escapedKey match {
       case statPattern(label, stat) => s"""$label{stat="$stat"}"""
       case _ => escapedKey
     }
-
-    escape3.replaceAllIn(parsedKey)
+    parsedKey
   }
```

Comment:
Nice, this is much easier to follow. Tiny nit -- the parsedKey val doesn't need to be assigned, and instead you could just return directly.  Something like:

```
private[admin] def formatKey(key: String) = {
  val escapedKey = disallowedChars.replaceAllIn(delimiter.replaceAllIn(key))
  escapedKey match {
    case statPattern(label, stat) => s"""$label{stat="$stat"}"""
    case _ => escapedKey
  }
}
```

Or perhaps even more concisely:

```
private[admin] def formatKey(key: String) =
  disallowedChars.replaceAllIn(delimiter.replaceAllIn(key)) match {
    case statPattern(label, stat) => s"""$label{stat="$stat"}"""
    case escapedKey => escapedKey
  }
```